### PR TITLE
crimson: keep do_osd_ops_params_t alive when call do_osd_ops

### PR DESCRIPTION
**Pull request by liu-chunmei**
otherwise stack-under-overflow

Signed-off-by: chunmei-liu <chunmei.liu@intel.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by athanatos** [comment]
retest this please

File: src/crimson/osd/pg.cc [hunk 1]
```diff
@@ -810,17 +810,19 @@ PG::do_osd_ops(
   ObjectContextRef obc,
   std::vector<OSDOp>& ops,
   const OpInfo &op_info,
-  const do_osd_ops_params_t& msg_params,
+  const do_osd_ops_params_t msg_params,
```
[the remark this answers is not in the document]
Comment (athanatos) [reply src/crimson/osd/pg.cc:813]: The only user seems to be InternalClientRequest via WatchTimeoutRequest.  get_do_osd_ops_params() constructs and returns it on the fly.  do_osd_ops should take it by && and move it into the do_with.

[the remark this answers is not in the document]
Comment (liu-chunmei) [reply src/crimson/osd/pg.cc]: done.

**Review by rzarzynski — APPROVED** [review]

**Comment by liu-chunmei** [comment]
jenkins retest please.

**Pull request merged**